open_city_profile/tests/conftest.py

```diff
@@ -18,13 +19,21 @@
 
 
 class GraphQLClient(GrapheneClient):
-    def execute(self, *args, **kwargs):
+    def execute(self, *args, context=None, **kwargs):
         """
         Custom wrapper on the execute method, allows adding the
         GQL DataLoaders middleware, since it has to be added to make
         the DataLoaders available through the context.
         """
-        return super().execute(*args, middleware=[GQLDataLoaders()], **kwargs)
+        if context is None:
+            context = RequestFactory().post("/graphql")
+
+        if not hasattr(context, "user") and hasattr(self, "user"):
+            context.user = self.user
+
+        return super().execute(
+            *args, context=context, middleware=[GQLDataLoaders()], **kwargs
```

Comment:
I know this execute call wasn't added in this PR, but should this be changed to something like
```python
kwargs.setdefault("middleware", [GQLDataLoaders()])
return super().execute(
    *args, context=context, **kwargs
)
```

So that the middleware can be customised if needed?

Reply:
Hmm, good question. If this was a "general use client", then perhaps it would be appropriate. As it's used only for a specific need in the tests, I don't think it makes much difference. I think it can be changed later when/if needed.